chainer/distributions/kl_divergence.py

```diff
@@ -0,0 +1,69 @@
+from chainer.distributions import normal
+from chainer.functions.math import exponential
+
+_KLDIVERGENCE = {}
+
+
+def register_kl(dist1, dist2):
+    """Decorator to register KL divergence function.
+
+    .. admonition:: Example
+
+       This test case runs only when `numpy>=1.10` is installed.
+
+       >>> from chainer import distributions
+       ... @register_kl(distributions.Normal, distributions.Normal)
+       ... def _kl_dist1_dist2(dist1, dist2):
+       ...     return KL
+
+    Args:
+        dist1(:class:`~chainer.Distribution`): Distribution to calculate KL
```

Comment:
Maybe better to write that it is the first (left) operand of the KL divergence because KL divergence is not symmetric against the operands. Ditto for `dist2`. It is also good to write the mathematical form here, too (like written in `kl_divergence` docstring below).